libs/s25main/ingameWindows/iwObservate.cpp

```diff
@@ -159,11 +129,14 @@ void iwObservate::Draw_()
         lastWindowPos = GetPos();
     }
 
-    if(followMovableId)
+    // If the object was valid previously, track it (checks IsValid() for us)
+    // If it returns false, it either expired or we lost it
+    if(lastValid && !pickedObject.Track(*view, following))
     {
-        if(!MoveToFollowedObj())
-            followMovableId = 0;
+        following = false;
+        UpdateFollowButton();
     }
+    lastValid = pickedObject.IsValid();
```

Comment:
should this be only set inside the condition? It can never become true by this, can it? --> `following = lastValid = false;`

Might even make sense to remove the 2nd overload of `Track` and do it here. But I guess the current version looks cleaner